programs/klend/src/utils/prices/stork.rs

```diff
@@ -10,50 +10,46 @@ use crate::LendingError;
 
 pub(super) fn get_stork_price_and_twap(
     stork_price_info: &AccountInfo,
-    stork_twap_info: &AccountInfo,
+    stork_twap_info: Option<&AccountInfo>,
 ) -> Result<TimestampedPriceWithTwap> {
-    let price_feed: TemporalNumericValueFeed = stork_sdk::temporal_numeric_value::TemporalNumericValueFeed::try_deserialize(
-        &mut &stork_price_info.data.borrow()[..]
-    ).map_err(|e| {
+    let price_feed = TemporalNumericValueFeed::try_from(stork_price_info).map_err(|e| {
         msg!("Error loading stork price feed: {:?}", e);
         error!(LendingError::PriceNotValid)
     })?;
-
     let feed_id = price_feed.id;
     let price = price_feed
         .get_latest_canonical_temporal_numeric_value_unchecked(&feed_id)
-        .ok()?;
-
+        .map_err(|e| {
+            msg!("Error getting stork price: {:?}", e);
+            error!(LendingError::PriceNotValid)
+        })?;
     validate_stork_price(price.quantized_value)?;
 
-    let twap: TemporalNumericValueFeed = stork_sdk::temporal_numeric_value::TemporalNumericValueFeed::try_deserialize(
-        &mut &stork_twap_info.data.borrow()[..]
-    )?;
+    let twap_tsp: Option<TimestampedPrice> = stork_twap_info.and_then(|twap_info| TemporalNumericValueFeed::try_from_slice(&mut &twap_info.data.borrow()[..])
+                .ok())
+                .and_then(|feed| feed.get_latest_canonical_temporal_numeric_value_unchecked(&feed_id).ok())
+                .filter(|t| validate_stork_price(t.quantized_value).is_ok())
+                .map(|t| t.into());
 
-    validate_stork_price(twap.quantized_value)?;
-
-    let twap = twap
-        .get_latest_canonical_temporal_numeric_value_unchecked(&feed_id)
-        .ok()?;
     Ok(TimestampedPriceWithTwap {
         price: price.into(),
-        twap: Some(twap.into()),
+        twap: twap_tsp,
     })
 }
 
 pub(super) fn validate_stork_price(
```

Comment:
What exactly are you checking here?

Reply:
Just checking that the price isn't zero. We don't necessarily have to validate the price, however another integrated oracle (pyth) was validating against zero.